frontend/src/modules/EditZing/Components/EditZing.tsx

```diff
@@ -10,10 +11,38 @@ import { GroupGrid } from 'EditZing/Components/GroupGrid'
 import { Student } from 'EditZing/Types/Student'
 import { DndProvider } from 'react-dnd'
 import { HTML5Backend } from 'react-dnd-html5-backend'
+import { getZingGroups, saveSwapStudent } from './Helpers'
+import { FetchedZing } from 'EditZing/Types/Student'
 
 export const EditZing = () => {
+  // get param that was set from history using location
+  const location = useLocation() as any
+  const params = location.state.params
+  const [zingId] = useState(params)
+
   const fakeStudentGroupsFromJson: Student[][] = require('EditZing/fakeData.json')
+  // full fetched zing object
+  const [zingData, setZingData] = useState<FetchedZing | null>(null)
+  // student groups parsed out from zingData into a Student[][]
   const [studentGroups, setStudentGroups] = useState(fakeStudentGroupsFromJson)
+  useEffect(() => {
+    async function fetchGroups(zingId: string) {
+      if (zingId !== undefined) {
+        var zingData = await getZingGroups(zingId)
+        setZingData(zingData)
+
+        // parse out groups into Student[][]
+        const zingGroup = zingData.group
+        let realData: Student[][] = []
+        for (var id in zingGroup) {
```

Comment:
I believe this can be const too.... if not I would use 'let'. you can google the differences but 'var' is fairly old.